src/backend/distributed/sql/downgrades/citus--9.5-1--9.4-1.sql

```diff
@@ -2,7 +2,27 @@
 
 SET search_path = 'pg_catalog';
 
-DROP FUNCTION create_citus_local_table(table_name regclass);
+-- Check if user have any citus local tables.
+-- If not, DROP create_citus_local_table UDF continue safely.
+-- Otherwise, raise an exception to stop the downgrade process.
+DO $$
+DECLARE
+    citus_local_table_count integer;
+BEGIN
+    SELECT COUNT(*) INTO citus_local_table_count
+    FROM pg_dist_partition WHERE repmodel != 't' AND partmethod = 'n';
```

Comment:
What could be an alternative here instead of checking repmodel and partmethod here? I am asking because what if we decide to change what identifies a `citus local table` in the future?

Reply:
We can first define a function that returns all citus local table OID's like: https://github.com/citusdata/citus/blob/3a73fba81082480b6d80cc348b944407f93bfe69/src/backend/distributed/metadata/metadata_cache.c#L3823-L3867


then we can define a udf to get the OID list for citus local tables using that function and return true if the length of that list is 0. And then we can use this UDF in migration script to error out.

Reply:
I think something like that could make more sense so that `the internal representation of a citus local table` is not hard coded here. 

The UDF could be as simple as `has_citus_local_table`, or if we choose to undistribute them we could return their names and try to undistribute all of them.

Reply:
I will introduce a similar function for citus local tables after merging the other pr (https://github.com/citusdata/citus/pull/4176).